src/test/java/com/spartasystems/holdmail/domain/MessageContentPartTest.java

```diff
@@ -0,0 +1,171 @@
+
+        MessageContentPart messageContentPart = new MessageContentPart();
+        assertThat(messageContentPart.getHeaders()).isEmpty();
+
+        messageContentPart.setHeader("a", "aval");
+        assertThat(messageContentPart.getHeaders()).isEqualTo(of("a", "aval"));
+
+        messageContentPart.setHeader("b", "bval");
+        assertThat(messageContentPart.getHeaders()).isEqualTo(of("a", "aval", "b", "bval"));
+
+        messageContentPart.setHeader("b", "newBVal");
+        assertThat(messageContentPart.getHeaders()).isEqualTo(of("a", "aval", "b", "newBVal"));
+    }
+
+    @Test
+    public void shouldGetContentType() throws Exception {
+
+        MessageContentPart messageContentPart = new MessageContentPart();
+        assertThat(messageContentPart.getContentType()).isNull();
+
+        messageContentPart.setHeader(CONTENT_TYPE, "magic-type");
+        assertThat(messageContentPart.getContentType()).isEqualTo("magic-type");
+
+    }
+
+    @Test
+    public void shouldGetIsHTML() throws Exception {
+
+        MessageContentPart messageContentPart = new MessageContentPart();
+        assertThat(messageContentPart.isHTML()).isFalse();
+
+        messageContentPart.setHeader(CONTENT_TYPE, "text/plain");
+        assertThat(messageContentPart.isHTML()).isFalse();
+
+        messageContentPart.setHeader(CONTENT_TYPE, "text/html");
+        assertThat(messageContentPart.isHTML()).isTrue();
+    }
+
+    @Test
+    public void shouldGetIsText() throws Exception {
+
+        MessageContentPart messageContentPart = new MessageContentPart();
+        assertThat(messageContentPart.isText()).isFalse();
+
+        messageContentPart.setHeader(CONTENT_TYPE, "text/html");
+        assertThat(messageContentPart.isText()).isFalse();
+
+        messageContentPart.setHeader(CONTENT_TYPE, "text/plain");
+        assertThat(messageContentPart.isText()).isTrue();
+    }
+
+    @Test
+    public void shouldHaveContentId() throws Exception {
+
+        MessageContentPart messageContentPart = new MessageContentPart();
+        assertThat(messageContentPart.hasContentId("thing")).isFalse();
+
+        messageContentPart.setHeader(CONTENT_ID, "");
+        assertThat(messageContentPart.hasContentId("thing")).isFalse();
+
+        messageContentPart.setHeader(CONTENT_ID, "gosh darnit bobby");
+
+        assertThat(messageContentPart.hasContentId("propane")).isFalse();
+        assertThat(messageContentPart.hasContentId("gosh")).isTrue();
+        assertThat(messageContentPart.hasContentId("darnit")).isTrue();
+        assertThat(messageContentPart.hasContentId("bobby")).isTrue();
+
+    }
+
+    @Test
+    public void shouldSetContent() throws Exception {
+
+        byte[] input = NON_ASCII_STR.getBytes();
+
+        MessageContentPart messageContentPart = new MessageContentPart();
+
+        messageContentPart.setContent(new ByteArrayInputStream(input));
+
+        byte[] actual = Whitebox.getInternalState(messageContentPart, "content");
```

Comment:
We should consider whether this is a sign of improving design due to testability issues.  I can understand the problem with exposing an array for its mutability, at the same time it may make sense always just provide that input stream and verify the value of the stream over the internal byte[].